hi-ml-multimodal/src/health_multimodal/image/model/model.py

```diff
@@ -212,5 +214,18 @@ def get_encoder_output_dim(module: torch.nn.Module) -> int:
     x = torch.rand((1, 3, 448, 448)).to(device)
 
     # Extract the number of output feature dimensions
-    representations = module(x)
+    with restore_training_mode(module):
+        module.eval()
+        representations = module(x)
     return representations.shape[1]
+
+
+@contextmanager
+def restore_training_mode(module: nn.Module) -> Generator[None, None, None]:
```

Comment:
@fepegar This method could be useful in other codebases. Wondering if it should be placed elsewhere, instead of in image/model.

Reply:
It should, I agree. But the import from `hubconf.py` is a bit brittle as it's designed so only `torch` and `torchvision` are needed, so I'd rather work on that on a different PR so we can unblock @markpinnock's PR (https://github.com/microsoft/hi-ml/pull/807). Does that make sense?

Reply:
I see, this sounds good.